base/stats.go

```diff
@@ -748,41 +840,68 @@ type QueryStat struct {
 	QueryTime       *SgwIntStat
 }
 
-func (s *SgwStats) NewDBStats(name string, deltaSyncEnabled bool, importEnabled bool, viewsEnabled bool, queryNames ...string) *DbStats {
+func (s *SgwStats) NewDBStats(name string, deltaSyncEnabled bool, importEnabled bool, viewsEnabled bool, queryNames ...string) (*DbStats, error) {
 	s.dbStatsMapMutex.Lock()
 	defer s.dbStatsMapMutex.Unlock()
 	s.DbStats[name] = &DbStats{
 		dbName: name,
 	}
 
 	// These have a pretty good chance of being used so we'll initialise these for every database stat struct created
-	s.DbStats[name].initCacheStats()
-	s.DbStats[name].initCBLReplicationPullStats()
-	s.DbStats[name].initCBLReplicationPushStats()
-	s.DbStats[name].initDatabaseStats()
-	s.DbStats[name].initSecurityStats()
+	err := s.DbStats[name].initCacheStats()
+	if err != nil {
+		return nil, err
+	}
+	err = s.DbStats[name].initCBLReplicationPullStats()
+	if err != nil {
+		return nil, err
+	}
+	err = s.DbStats[name].initCBLReplicationPushStats()
+	if err != nil {
+		return nil, err
+	}
+	err = s.DbStats[name].initDatabaseStats()
+	if err != nil {
+		return nil, err
+	}
+	err = s.DbStats[name].initSecurityStats()
+	if err != nil {
+		return nil, err
+	}
 
 	if deltaSyncEnabled {
-		s.DbStats[name].InitDeltaSyncStats()
+		err = s.DbStats[name].InitDeltaSyncStats()
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	if importEnabled {
-		s.DbStats[name].InitSharedBucketImportStats()
+		err = s.DbStats[name].InitSharedBucketImportStats()
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	if viewsEnabled {
-		s.DbStats[name].InitQueryStats(
+		err := s.DbStats[name].InitQueryStats(
 			true,
 			queryNames...,
 		)
+		if err != nil {
+			return nil, err
+		}
 	} else {
-		s.DbStats[name].InitQueryStats(
+		err := s.DbStats[name].InitQueryStats(
 			false,
 			queryNames...,
 		)
+		if err != nil {
+			return nil, err
+		}
 	}
```

Comment:
Can maybe avoid an error check here. Inside `ifs` rather than doing `err :=` do `err =` and then can error check outside of the if else